pkg/controller/jivavolume/jivavolume_controller.go

```diff
@@ -0,0 +1,709 @@
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+
+	// Fetch the JivaVolume instance
+	instance := &jv.JivaVolume{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	switch instance.Status.Phase {
+	case jv.JivaVolumePhaseCreated, jv.JivaVolumePhaseSyncing:
+		return reconcile.Result{}, r.getAndUpdateVolumeStatus(instance, reqLogger)
+	case jv.JivaVolumePhaseDeleting:
+		reqLogger.Info("start tearing down jiva components", "JivaVolume: ", instance)
+		return reconcile.Result{}, nil //r.teardownJivaComponents(instance, reqLogger)
+	case jv.JivaVolumePhasePending, jv.JivaVolumePhaseFailed:
+		reqLogger.Info("start bootstraping jiva components", "JivaVolume: ", instance)
+		return reconcile.Result{}, r.bootstrapJiva(instance, reqLogger)
+	}
+
+	reqLogger.Info("start bootstraping jiva components")
+
+	return reconcile.Result{}, r.bootstrapJiva(instance, reqLogger)
+}
+
+func (r *ReconcileJivaVolume) finally(err error, cr *jv.JivaVolume, reqLog logr.Logger) {
+	if err != nil {
+		cr.Status.Phase = jv.JivaVolumePhaseFailed
+	} else {
+		cr.Status.Phase = jv.JivaVolumePhaseSyncing
+	}
+
+	if err := r.updateJivaVolume(cr); err != nil {
+		reqLog.Error(err, "failed to update JivaVolume phase")
+	}
+}
+
+// 1. Create controller svc
+// 2. Create controller deploy
+// 3. Create replica statefulset
+func (r *ReconcileJivaVolume) bootstrapJiva(cr *jv.JivaVolume, reqLog logr.Logger) (err error) {
+	defer r.finally(err, cr, reqLog)
+
+	for _, f := range installFuncs {
+		if err = f(r, cr, reqLog); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// TODO: add logic to create disruption budget for replicas
+func createReplicaPodDisruptionBudget(r *ReconcileJivaVolume, cr *jv.JivaVolume, reqLog logr.Logger) error {
+	min, err := strconv.Atoi(cr.Spec.ReplicationFactor)
+	if err != nil {
+		return fmt.Errorf("failed to convert to int, err: %v", err)
+	}
+	pdbObj := &policyv1beta1.PodDisruptionBudget{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PodDisruptionBudget",
+			APIVersion: "policyv1beta1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name + "-pdb",
+			Namespace: cr.Namespace,
+		},
+		Spec: policyv1beta1.PodDisruptionBudgetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: defaultReplicaLabels(cr.Spec.PV),
+			},
+			MinAvailable: &intstr.IntOrString{
+				Type:   intstr.Int,
+				IntVal: int32(min/2 + 1),
+			},
+		},
+	}
+
+	found := &policyv1beta1.PodDisruptionBudget{}
```

Comment:
Can we please rename this variable everywhere, to corresponding type.

Reply:
done